### PR TITLE
docs: Update APM Server config

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

The goal of this PR is to combine APM Server binary and Fleet-managed config options into one configuration section.

The following topics are complete:

[**Configure**](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuring-howto-apm-server.html)
- [x] [General configuration options](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuration-process.html)
- [x] [Anonymous authentication](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuration-anonymous.html)
- [x] [APM agent authorization](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/_apm_agent_authorization.html)
- [x] [APM agent configuration](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configure-agent-config.html)
- [x] [Instrumentation](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuration-instrumentation.html)
- [x] [Kibana endpoint](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/setup-kibana-endpoint.html)
- [x] [Logging](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuration-logging.html)
- [x] [Output](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuring-output.html)
- [x] [Project paths](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuration-path.html)
- [x] [Real User Monitoring (RUM)](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuration-rum.html)
- [ ] [SSL/TLS settings](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/configuration-ssl-landing.html)
- [x] [Tail-based sampling](https://apm-server_10842.docs-preview.app.elstc.co/guide/en/apm/guide/master/_tail_based_sampling.html)

### Help

I'm struggling with the SSL/TLS section

### CODE REVIEW

File: docs/configure/general.asciidoc [hunk 1]
```diff
@@ -6,210 +6,208 @@ General APM Server configuration options.
...
 Maximum permitted duration for reading an entire request.
-Defaults to 30 seconds.
+Defaults to `3600s` (3600 seconds). (text)
+
+|====
+| APM Server binary | `apm-server.read_timeout`
+| Fleet-managed     | Maximum duration for reading an entire request
+|====
+
+// REVIEWER: Default 3600 seconds (FM) or 30 seconds (binary)?
```
Comment (axw) [inline docs/configure/general.asciidoc:66]: I think we should treat the integration config as canonical for now (so 3600s). I've opened an issue to align the defaults: https://github.com/elastic/apm-server/issues/10843

File: docs/configure/general.asciidoc [hunk 2]
```diff
@@ -6,210 +6,208 @@ General APM Server configuration options.
...
 Maximum duration in seconds before releasing resources when shutting down the server.
-Defaults to 5 seconds.
+Defaults to `30s` (30 seconds). (text)
+
+|====
+| APM Server binary | `apm-server.shutdown_timeout`
+| Fleet-managed     | Maximum duration before releasing resources when shutting down
+|====
+
+// REVIEWER: Default 30 seconds (FM) or 5 seconds (binary)?
```
Comment (axw) [inline docs/configure/general.asciidoc:90]: 30s - that's the default for the binary too. Looks like apm-server.yml is out of date, I'll fix that.

**Review by axw — COMMENTED** [review]
Looks good!

**Comment by simitt** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/configure/outputs/logstash.asciidoc [hunk 3]
```diff
@@ -5,6 +5,12 @@
 <titleabbrev>{ls}</titleabbrev>
 ++++
 
+****
+image:./binary-yes-fm-no.svg[supported deployment methods]
+
+The {ls} output is not yet supported by {fleet}-managed APM Server.
+****
+
```
Comment (simitt) [inline docs/configure/outputs/logstash.asciidoc:13]: Please change `Kibana output` to `Kibana configuration` in https://github.com/elastic/apm-server/pull/10842/files#diff-93a49822f5cc2506c396df37910cb6d828a9059a6fb56c12d711af977b4de355R48

**Comment by bmorelli25** [comment]
Working on these...

```
05:41:57 INFO:build_docs:Bad cross-document links:
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/beats/auditbeat/master/configuration-instrumentation.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/beats/filebeat/master/configuration-instrumentation.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/beats/functionbeat/master/configuration-instrumentation.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/beats/heartbeat/master/configuration-instrumentation.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/beats/metricbeat/master/configuration-instrumentation.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/beats/packetbeat/master/configuration-instrumentation.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/beats/winlogbeat/master/configuration-instrumentation.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
05:41:57 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/observability/master/monitor-kubernetes.html contains broken links to:
05:41:57 INFO:build_docs:   - en/apm/guide/master/input-apm.html
```

File: docs/configure/outputs/logstash.asciidoc (hunk 3 above)
Comment (bmorelli25) [reply docs/configure/outputs/logstash.asciidoc:13]: done

File: docs/configure/general.asciidoc [hunk 4]
```diff
@@ -1,215 +1,228 @@
...
-=== `auth.secret_token`
-Authorization token for sending data to the APM server.
-If a token is set, the agents must send it in the following format:
-Authorization: Bearer <secret-token>.
-The token is not used for RUM endpoints. By default, no authorization token is set.
+== Custom HTTP response headers
+Custom HTTP headers to add to HTTP responses. Useful for security policy compliance. (text)
+
+|====
+| APM Server binary | N/A
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/configure/general.asciidoc]: All feedback below this comment was addressed in 4b42e701411599ed7564ce1e98e3daf282f11d05

File: docs/configure/general.asciidoc [hunk 5]
```diff
@@ -1,215 +1,228 @@
...
 
+****
+image:./binary-yes-fm-yes.svg[supported deployment methods]
+
+Most options on this page are supported by all APM Server deployment methods.
+****
+
 General APM Server configuration options.
 
 include::./tab-widgets/general-config-widget.asciidoc[]
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/configure/general.asciidoc:12]: This comment and the comment below it have been addressed in b0ed9ba

**Review by simitt — APPROVED** [review]
Great docs changes!

**Pull request merged**